### PR TITLE
RTT Logging Setup

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary

Setup RTT logging to get `printf` style debugging working on embedded and tests.

### Changelist 

- Setup `LOG_...` debug macros (one for info, warnings, and errors). 
- Setup RTT configuration in VS Code.

### Testing Done

- Tested on embedded on F4, H7, and tests

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

**Review by will-chaba — APPROVED** [review]
TLDR.
